cmake/KokkosConfigCommon.cmake.in

```diff
@@ -306,3 +306,38 @@ FUNCTION(kokkos_compilation)
         ENDFOREACH()
     ENDIF()
 ENDFUNCTION()
+
+# this function is provided to easily set which files use the same programming language as Kokkos:
+#
+#       SOURCE      --> specific source files
+#
+FUNCTION(kokkos_set_source_language LANG_SOURCE)
+
+      IF(NOT DEFINED Kokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE)
+        MESSAGE(FATAL_ERROR "Using language features requires Kokkos to be build with Kokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE")
+      ENDIF()
+
+      IF(LANG_SOURCE)
+        MESSAGE(DEBUG "KOKKOS: Setting language to ${Kokkos_COMPILE_LANGUAGE} for SOURCE ${LANG_SOURCE}")
+        SET_PROPERTY(SOURCE ${LANG_SOURCE} PROPERTY LANGUAGE ${Kokkos_COMPILE_LANGUAGE})
+      ENDIF()
+
+ENDFUNCTION()
+
+# this function is provided to easily set the language specific properties of a target to the ones exported by kokkos:
+#
+#       TARGET      --> specific TARGET
+#
+FUNCTION(kokkos_set_target_properties PROP_TARGET)
+
+      IF(NOT DEFINED Kokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE)
+        MESSAGE(FATAL_ERROR "Using language features requires Kokkos to be build with Kokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE")
+      ENDIF()
+
+      IF(PROP_TARGET)
```

Comment:
I expect this should be `IF(TARGET PROP_TARGET)` and same observations as above, I would be fine just directly calling `set_target_properties` without the guard and think that we should consider making it legal to call w/o compile as cmake language.

Reply:
I am even tempted to suggest we make it link against kokkos as well.  Thoughts?

Reply:
yeah...it would be convenient. I did not do that because I thought it is more expressive having the link command outside